### PR TITLE
Add DSLs

### PR DESCRIPTION
**Pull request by nivgsk**
Add DSL files (modified versions of [Kong](https://github.com/kenshoo/kong/tree/master/buildScripts/jenkins/dsl)'s) to support automating the MemSQL automation base image via Jenkins.

### CODE REVIEW

File: buildScripts/jenkins/dsl/pull_request.groovy [hunk 1]
```diff
@@ -0,0 +1,100 @@
+def NAME = "memsql-base-automation-image-builder"
+def EMAIL = "_devmicrocosm@kenshoo.com"
+def JOB_NAME = "${NAME}-pull-request"
+
+job(JOB_NAME) {
+    label("microcosm-ubuntu-base")
```
Comment (giladgurvich) [inline buildScripts/jenkins/dsl/pull_request.groovy:6]: 👍

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 .idea
-.vscode
\ No newline at end of file
+.vscode
+lib
```
Comment (giladgurvich) [inline .gitignore:3]: was this added automatically?

**Review by giladgurvich — COMMENTED** [review]
looks good :)
do we want to generate the jobs from this branch or merge?

File: .gitignore (hunk 2 above)
Comment (nivgsk) [reply .gitignore:3]: @giladgurvich no, I let IJ download the groovy jdk for syntax highlighting and it added a bunch of files to this directory, so I added it to gitignore

**Review by giladgurvich — APPROVED** [review]

**Pull request merged**